File: javascript/src/examples/FeatureCollection/ReverseMask.js
// Reverse mask a region.
//
// Create an image that masks off the interior of the specified polygon.

var fc = ee.FeatureCollection('ft:1Ec8IWsP8asxN-ywSqgXWMuBaxI6pPaeh6hC64lA')
    .filter(ee.Filter().eq('ECO_NAME', 'Great Basin shrub steppe'));

// Fill and outline the polygons in two colors
var region = ee.Image(0).byte()
    .paint(fc, 2)       // Fill with 2
    .paint(fc, 1, 2);   // Outline with 1, width 2.

// Mask off everything that matches the fill color.
var result = region.updateMask(region.neq(2));

Map.addLayer(result, {
  palette: '000000,FF0000',
  max: 1,
  opacity: 0.5
});
Map.setCenter(-100, 40, 4);

File: python/examples/py/FeatureCollection/reverse_mask.py
"""Reverse mask a region.

Create an image that masks everything except for the specified polygon.
"""

import ee
import ee.mapclient

ee.Initialize()
ee.mapclient.centerMap(-115, 39, 5)

fc = (ee.FeatureCollection('RESOLVE/ECOREGIONS/2017')
      .filter(ee.Filter().eq('ECO_NAME', 'Great Basin shrub steppe')))

# Start with a black image.
empty = ee.Image(0).toByte()
# Fill and outline the polygons in two colors
filled = empty.paint(fc, 2)
both = filled.paint(fc, 1, 5)
# Mask off everything that matches the fill color.
result = both.mask(filled.neq(2))

ee.mapclient.addToMap(result, {
    'palette': '000000,FF0000',
    'max': 1,
    'opacity': 0.5
})
